File: halcyon-web/src/main/resources/static/js/controller/account/sysRoleController.js
//控制层
app.controller("sysRoleController", function ($scope, $controller, sysRoleService, hrEmployeeService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.sysRole = {};
    }
    $scope.initForUpdate = function (id) {
        sysRoleService.queryById(id).success(function (data) {
            $scope.sysRole = data;
        });
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        sysRoleService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.save = function () {
        sysRoleService.save($scope.sysRole).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    }

    function deleteRows() {
        sysRoleService.delete($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }

});

File: halcyon-base/src/main/java/com/marshal/halcyon/base/account/controller/SysRoleController.java
package com.marshal.halcyon.base.account.controller;

import com.marshal.halcyon.base.account.entity.SysRole;
import com.marshal.halcyon.base.account.entity.SysRoleFunction;
import com.marshal.halcyon.base.account.entity.SysUser;
import com.marshal.halcyon.base.account.service.SysRoleFunctionService;
import com.marshal.halcyon.base.account.service.SysRoleService;
import com.marshal.halcyon.base.account.service.SysUserService;
import com.marshal.halcyon.base.function.service.SysFunctionService;
import com.marshal.halcyon.core.component.SessionContext;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.core.entity.ResponseData;
import com.marshal.halcyon.core.util.RequestHelper;
import com.marshal.halcyon.core.util.ResponseUtil;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletRequest;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/sys/role")
public class SysRoleController extends BaseController {

    @Autowired
    SysRoleService sysRoleService;

    @Autowired
    SysFunctionService sysFunctionService;

    @Autowired
    SysRoleFunctionService sysRoleFunctionService;

    @RequestMapping("/query")
    public ResponseData query(@RequestBody SysRole condition, int pageNum, int pageSize) {
        List<SysRole> list = sysRoleService.select(condition, pageNum, pageSize);
        return new ResponseData(list);
    }

    @RequestMapping("/submit")
    public ResponseData save(@RequestBody SysRole sysRole) {
        if (getValidator().hasError(sysRole)) {
            return new ResponseData(false, getValidator().getErrors(sysRole));
        }
        if (sysRoleService.save(sysRole) == 0) {
            return ResponseUtil.responseErr();
        }
        return ResponseUtil.responseOk();
    }

    @RequestMapping("/remove")
    public ResponseData delete(@RequestParam("selectedIds") Long[] selectedIds) {
        sysRoleService.batchDelete(selectedIds);
        return new ResponseData(true, "删除成功");
    }

    /**
     * 获取制定角色的菜单分配详情
     *
     * @param request
     * @return
     */
    @RequestMapping("/functionAssignList")
    public List<Map> selectRoleFunctionAssign(HttpServletRequest request,
                                              @RequestParam Long roleId) {
        return sysFunctionService.selectRoleFunctionAssignList(roleId);
    }

    @PostMapping("/functionAssign")
    public ResponseData functionAssign(@RequestBody List<SysRoleFunction> sysRoleFunctions) {
        sysFunctionService.updateSysRoleFunctions(sysRoleFunctions);
        return ResponseUtil.responseOk();
    }
}
